### PR TITLE
Test(eos_designs): Create eos_designs deprecated vars molecule unit test

### LINKED ISSUE #2947: Test: create eos_designs deprecated vars molecule unit test
Create a new molecule scenario containing all deprecated variables.

Include all variables that support convert_dicts.

Test corner case in convert_dicts with keys that have no value:
```yaml
l3leaf:
  node_groups:
    - group: DC1_LEAF1
       structured_config:
          sflow:
            vrfs:
              OOB:
                destinations:
                  192.168.200.10:
                  10.0.200.90:
```

### PR DESCRIPTION
**Pull request by Vibhu-gslab**
## Change Summary

Create a new molecule scenario containing all deprecated variables for eos_designs.

## Related Issue(s)

Fixes #2947 

## Component(s) name

`arista.avd.eos_designs`

### User Checklist

- N/A

### Repository Checklist

- [ ] My code has been rebased from devel before I start
- [ ] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [ ] My change requires a change to the documentation and documentation have been updated accordingly.
- [ ] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/host_vars/connected_endpoints_keys_key.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+l3leaf:
+  defaults:
+    loopback_ipv4_pool: 192.168.255.0/24
+    vtep_loopback_ipv4_pool: 192.168.254.0/24
+  nodes:
+    - name: connected_endpoints_keys_key
+      id: 101
+      bgp_as: 101
+
+servers:
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/host_vars/connected_endpoints_keys_key.yml]:
```suggestion
# Testing "servers" as dict of dict
# Dict type is deprecated in 4.0.0. To be removed in 5.0.0
servers:
```

**Review by ClausHolbechArista — COMMENTED** [review]
Please look into collapsing these to a single ansible host "host1" in the molecule scenario, and then use a folder "host1" under hostvars.
See the `eos_cli_config_gen_deprecated_vars` for example.

Maybe the mpls stuff have to be under a "host2" since it is contradicting the settings under "host1"

File: ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/host_vars/connected_endpoints_keys_key.yml (hunk 1 above)
Comment (Vibhu-gslab) [reply ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/host_vars/connected_endpoints_keys_key.yml]: Done!

File: ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/group_vars/all.yml [hunk 2]
```diff
@@ -4,3 +4,17 @@ is_for_test: true
...
+  servers:
+    type: server
+  firewalls:
+    type: firewall
+  routers:
+    type: router
+  load_balancers:
+    type: load_balancer
+  storage_arrays:
+    type: storage_array
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/group_vars/all.yml]: I think you can move this to under host1. For host2 it can just use the defaults which are the same.

File: ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/host_vars/host1/connected_endpoints_keys_key.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+type: l3leaf
+l3leaf:
+  defaults:
+    loopback_ipv4_pool: 192.168.255.0/24
+    vtep_loopback_ipv4_pool: 192.168.254.0/24
+  nodes:
+    - name: connected_endpoints_keys_key
+      id: 101
+      bgp_as: 101
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/host_vars/host1/connected_endpoints_keys_key.yml]: This config should only be in one file, so remove it from all the others.

File: ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/host_vars/connected_endpoints_keys_key.yml (hunk 1 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/host_vars/connected_endpoints_keys_key.yml]: Descriptions like this should be put in above _every_ dict-of-dict with the wildcard keys, so we can quickly identify what is being tested.

File: ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/hosts.yml
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/hosts.yml:1]: Please revert back to the hosts.yml format instead of hosts.ini.

File: ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/host_vars/connected_endpoints_keys_key.yml (hunk 1 above)
Comment (Vibhu-gslab) [reply ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/host_vars/connected_endpoints_keys_key.yml]: Done

File: ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/hosts.yml
Comment (Vibhu-gslab) [reply ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/hosts.yml:1]: Done

File: ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/host_vars/host1/connected_endpoints_keys_key.yml (hunk 3 above)
Comment (Vibhu-gslab) [reply ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/host_vars/host1/connected_endpoints_keys_key.yml]: Done

File: ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/group_vars/all.yml (hunk 2 above)
Comment (Vibhu-gslab) [reply ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/group_vars/all.yml]: Done

File: ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/host_vars/host1/port-channel-short-esi.yml [hunk 4]
```diff
@@ -16,6 +6,8 @@ servers:
         switch_ports: [ Ethernet10, Ethernet10]
         switches: [ port-channel-short-esi, Leaf1]
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/host_vars/host1/port-channel-short-esi.yml]: The test output does not contain these port-channels because `switches` need to contain `host1` + one other dummy name (keep Leaf1). Same below
```suggestion
        switches: [ host1, Leaf1]
```

File: ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/host_vars/host2/network_services_keys_name.yml [hunk 5]
```diff
@@ -0,0 +1,67 @@
+---
+type: pe
+design:
+  type: mpls
+
+pe:
+  defaults:
+    loopback_ipv4_pool: 10.42.0.0/24
+    bgp_as: 42
+    id: 42
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/host_vars/host2/network_services_keys_name.yml]: Please move this to `node_type.yml`

File: ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/host_vars/host2/network_services_keys_name.yml
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/host_vars/host2/network_services_keys_name.yml:1]: Please rename this file to `tenants.yml`

File: ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/host_vars/host1/port-channel-short-esi.yml (hunk 4 above)
Comment (Shivani-chourasiya) [reply ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/host_vars/host1/port-channel-short-esi.yml]: Fixed

File: ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/host_vars/host2/network_services_keys_name.yml (hunk 5 above)
Comment (Shivani-chourasiya) [reply ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/host_vars/host2/network_services_keys_name.yml]: we have type `l3leaf` in node_type.yml, did you mean `node_type_keys.yml`?

File: ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/host_vars/host2/network_services_keys_name.yml
Comment (Shivani-chourasiya) [reply ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/host_vars/host2/network_services_keys_name.yml:1]: Fixed

File: ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/host_vars/host2/network_services_keys_name.yml (hunk 5 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/host_vars/host2/network_services_keys_name.yml]: This is for host2. I don't think you have l3leaf anywhere for host2. If so it would not work well. A key should only be set in _one_ file within the host_vars.

File: ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/host_vars/host1/connected_endpoints_keys_key.yml
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/host_vars/host1/connected_endpoints_keys_key.yml:1]: Rename this to `routers.yml`

File: ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/host_vars/host2/network_services_keys_name.yml (hunk 5 above)
Comment (Shivani-chourasiya) [reply ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/host_vars/host2/network_services_keys_name.yml]: ohh, so we need to keep a file node_type.yml in host2 with this config?

File: ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/host_vars/host1/port-channel-short-esi.yml [hunk 6]
```diff
@@ -26,6 +18,8 @@ servers:
         mtu: 1601
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/host_vars/host1/port-channel-short-esi.yml:18]: I cannot insert the comment in the right line, but the `switches` key in line 17 should also be updated with `host1` to lead to any config.

File: ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/host_vars/host2/network_services_keys_name.yml (hunk 5 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/host_vars/host2/network_services_keys_name.yml]: One node_type.yml for host1 and another for host2. This comment is only about the file for host2.

Comment (Shivani-chourasiya) [reply ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/host_vars/host2/network_services_keys_name.yml]: Got it!

File: ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/host_vars/host1/connected_endpoints_keys_key.yml
Comment (Shivani-chourasiya) [reply ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/host_vars/host1/connected_endpoints_keys_key.yml:1]: done

File: ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/host_vars/host1/port-channel-short-esi.yml (hunk 6 above)
Comment (Shivani-chourasiya) [reply ansible_collections/arista/avd/molecule/eos_designs_deprecated_vars/inventory/host_vars/host1/port-channel-short-esi.yml:18]: fixed

**Review by ClausHolbechArista — APPROVED** [review]
